addons/mail/static/src/core/common/thread_model.js

```diff
@@ -964,31 +964,44 @@ export class Thread extends Record {
         );
     }
 
+    async renameChannel(name) {
+        this.name = newName;
+        await this.store.env.services.orm.call(
+            "discuss.channel",
+            "channel_rename",
+            [[this.id]],
+            { name }
+        );
+    }
+
+    async setCustomChannelName(name) {
+        this.custom_channel_name = newName;
+        await this.store.env.services.orm.call(
+            "discuss.channel",
+            "channel_set_custom_name",
+            [[this.id]],
+            { name }
+        );
+    }
+
+    prepareName(name) {
+        return name.trim();
+    }
+
+
     /** @param {string} name */
     async rename(name) {
-        const newName = name.trim();
+        const newName = this.prepareName()
         if (
             newName !== this.displayName &&
             ((newName && this.channel_type === "channel") ||
                 this.channel_type === "chat" ||
                 this.channel_type === "group")
         ) {
             if (this.channel_type === "channel" || this.channel_type === "group") {
-                this.name = newName;
-                await this.store.env.services.orm.call(
-                    "discuss.channel",
-                    "channel_rename",
-                    [[this.id]],
-                    { name: newName }
-                );
+                await this.renameChannel(newName);
             } else if (this.channel_type === "chat") {
-                this.custom_channel_name = newName;
-                await this.store.env.services.orm.call(
-                    "discuss.channel",
-                    "channel_set_custom_name",
-                    [[this.id]],
-                    { name: newName }
-                );
+                await this.setCustomChannelName(newName);
```

Comment:
If this branching was `this.isChatChannel && this.channel_type !== "group"` then it would be taken into account for both Livechat and WhatsApp, which is the desired improvement of this task.